src/linear_algebra/multilinear.lean

```diff
@@ -515,6 +516,135 @@ protected def pi_ring_equiv [fintype ι]  : M₂ ≃ₗ[R] (multilinear_map R (
 
 end comm_ring
 
+section algebra
+
+variables (S : Type u) [comm_semiring S]
+variables {N : Type u} [semiring N] [algebra S N]
+variable {q : ℕ}
+variable (ν : fin q → N)
+
+@[priority 100] instance algebra_to_semimodule : semimodule S N := algebra.to_semimodule
+
+/--
+Auxiliary function used to define the multilinear map `algebra_prod`.
+-/
+def algebra_prod_aux : N := list.prod $ list.map ν (list.fin_range q)
+
+lemma algebra_prod_split_aux (ν : fin q.succ → N) :
+  algebra_prod_aux ν = (ν 0) * algebra_prod_aux (λ i : fin q, ν i.succ) :=
+begin
+  have key : list.fin_range q.succ =
+  0 :: list.map fin.succ (list.fin_range q) :=
+  begin
+    unfold list.fin_range,
+    rw [list.map_pmap],
+    simp_rw [list.range_succ_eq_map],
+    rw [list.pmap],
+    congr' 1,
+    rw list.pmap_map,
+    apply list.pmap_congr,
+    intros, refl,
+  end,
+  unfold algebra_prod_aux,
+  rw [key, list.map_cons, list.prod_cons],
+  simp,
+end
+
+/--
+The multilinear map sending a vector of elements of a semiring to their product.
+-/
+def algebra_prod : multilinear_map S (λ i : fin q, N) N :=
```

Comment:
This should make sense for any fintype, not only `fin q`.

Reply:
This is precisely `multilinear_map.mk_pi_algebra_fin`, I think - so is no longer needed.